Enable Nuxt Bridge integration and add composables

Strift opened the pull request:
Hello,

I love Supabase and the integration you built for Vue & Nuxt.

I saw that your other repository had an open issue for [Nuxt 3 support](https://github.com/supabase-community/nuxt-supabase/issues/14) and I wanted to try and help.

Since the Nuxt module is using this Vue plugin, I added some new composables in here.

Furthermore, to enable the Bridge integration, I needed the Vue 2 app to provide `supabase`, so I added that as well. I _think_ the mixin is the Vue 2 way of doing this, but I'm not sure.

I will also open a PR on the `nuxt-supabase` repository (will edit when done). I think this will make more sense in the light of it.

I didn't add tests but did a lot of manual testing.
The docs still needs to be updated, I'll be more than happy to do it if the implementation is deemed satisfying. :) 

PS: I'm pretty new to open-source and still learning TypeScript. So if there's anything that needs updating, just let me know!

sduduzog commented on .nvmrc:
```diff
@@ -0,0 +1 @@
+v14.17.3
```
Quick question, what impact does this file have for anyone who does not have this version on node installed, given they have nvm anyways?

Strift replied:
If you have NVM installed, you still need to run `nvm use` to use the version in `.nvmrc`. Unless you have your shell configured to auto-run the command, but that would still happen behind the scenes.

Now, if you don't have the correct version installed, this is the error message you would get:

```
Found '/home/strift/Code/OpenSource/vue-supabase/.nvmrc' with version <v14.17.3>
N/A: version "v14.17.3 -> N/A" is not yet installed.

You need to run "nvm install v4.9.1" to install it before using it.
```

sduduzog reviewed (changes requested):
@scottrobertson I've tested with my existing projects, both for Vue 2x and Vue 3x and all is well.

@Strift Please modify the package.json version, possibly to version 2.2.4 so that github actions pick up this PR as a possible release

Strift commented:
Hello @sduduzog, thanks for taking the time to review the PR!

Sorry I took a bit of time to get back to this, I've been sick the past week. :sweat_smile: 

I've updated the `package.json` version to `2.2.4` as requested. I also updated the URLs to the repository which were outdated.

Cheers,
